includes/event/event-repository.php

```diff
@@ -20,14 +20,36 @@ public function __construct( Attendee_Repository $attendee_repository ) {
 		$this->attendee_repository = $attendee_repository;
 	}
 
+	private function get_year_post_id( $year ) {
+		$year_post = get_page_by_path( $year, OBJECT, self::POST_TYPE );
+		if ( ! $year_post ) {
+			return wp_insert_post(
+				array(
+					'post_type'    => self::POST_TYPE,
+					'post_title'   => $year,
+					'post_name'    => $year,
+					'post_status'  => 'publish',
+					'post_content' => '',
+				)
+			);
+		}
+		return $year_post->ID;
+	}
+
 	public function insert_event( Event $event ) {
+		$post_parent = $this->get_year_post_id( $event->start()->utc()->format( 'Y' ) );
+		if ( is_wp_error( $post_parent ) ) {
+			return $post_parent;
+		}
+
 		$event_id_or_error = wp_insert_post(
 			array(
 				'post_type'    => self::POST_TYPE,
 				'post_name'    => $event->slug(),
 				'post_title'   => $event->title(),
 				'post_content' => $event->description(),
 				'post_status'  => $event->status(),
+				'post_parent'  => $post_parent,
```

Comment:
I think it would make sense to modify `test_create_event()` to make sure the post parent is set correctly.